src/components/PopoverWithMeasuredContent.js

```diff
@@ -59,138 +59,115 @@ const defaultProps = {
  * This way, we can shift the position of popover so that the content is anchored where we want it relative to the
  * anchor position.
  */
-class PopoverWithMeasuredContent extends Component {
-    constructor(props) {
-        super(props);
-
-        this.popoverWidth = lodashGet(this.props, 'popoverDimensions.width', 0);
-        this.popoverHeight = lodashGet(this.props, 'popoverDimensions.height', 0);
 
-        this.state = {
-            isContentMeasured: this.popoverWidth > 0 && this.popoverHeight > 0,
-            isVisible: false,
-        };
-
-        this.measurePopover = this.measurePopover.bind(this);
-    }
+function PopoverWithMeasuredContent(props) {
+    const {windowWidth, windowHeight} = useWindowDimensions();
+    const [popoverWidth, setPopoverWidth] = useState(props.popoverDimensions.width);
+    const [popoverHeight, setPopoverHeight] = useState(props.popoverDimensions.height);
+    const [isContentMeasured, SetIsContentMeasured] = useState(popoverWidth > 0 && popoverHeight > 0);
+    const [isVisible, SetIsVisible] = useState(false);
 
     /**
      * When Popover becomes visible, we need to recalculate the Dimensions.
      * Skip render on Popover until recalculations have done by setting isContentMeasured false as early as possible.
-     *
-     * @static
-     * @param {Object} props
-     * @param {Object} state
-     * @return {Object|null}
      */
-    static getDerivedStateFromProps(props, state) {
+    useEffect(() => {
         // When Popover is shown recalculate
-        if (!state.isVisible && props.isVisible) {
-            return {isContentMeasured: lodashGet(props, 'popoverDimensions.width', 0) > 0 && lodashGet(props, 'popoverDimensions.height', 0) > 0, isVisible: true};
+        if (!isVisible && props.isVisible) {
+            SetIsContentMeasured(lodashGet(props, 'popoverDimensions.width', 0) > 0 && lodashGet(props, 'popoverDimensions.height', 0) > 0);
+            SetIsVisible(true);
```

Comment:
I haven't tested this, but I wonder if doing this in parallel might result in a weird behavior since these calls are async. Before we used to set them at the same time, but now it's possible they will be out of sync. Not sure if that'll cause issues though

Reply:
I think it's better to avoid this, thanks for the clue!